merkledag/merkledag.go

```diff
@@ -460,16 +443,24 @@ func EnumerateChildrenAsync(ctx context.Context, ds DAGService, c *cid.Cid, visi
 			if inProgress == 0 && next == nil {
 				return nil
 			}
-		case nc := <-out:
-			if nc.Err != nil {
-				return nc.Err
+		case l := <-out:
+			n, err := ds.Get(ctx, l)
+			if err != nil {
+				return err
 			}
 
 			if next == nil {
-				next = nc.Node
-				send = feed
+				if len(n.Links()) > 0 {
+					next = n.Links()[0].Cid
+					for _, l := range n.Links()[1:] {
+						todobuffer = append(todobuffer, l.Cid)
+					}
+					send = feed
+				}
 			} else {
-				todobuffer = append(todobuffer, nc.Node)
+				for _, l := range n.Links() {
+					todobuffer = append(todobuffer, l.Cid)
```

Comment:
Same (https://github.com/ipfs/go-ipfs/pull/3598/files#r96632953) applies here.

The allocation of the additional array will probably be removed by optimization step in golang.